### PR TITLE
Give credit to original `WeakKeyIdDict` authors

### PR DESCRIPTION
**Pull request by haberdashPI**

### CODE REVIEW

File: LICENSE.md [hunk 1]
```diff
@@ -19,14 +22,3 @@ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
-
-The source implementation was originally copied from that found in `AbstractAlgebra.jl`
```
Comment (ericphanson) [inline LICENSE.md:23]: I think we should probably keep this? Given that it says "Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer."

maybe there are other licenses we should copy in as well?

File: README.md [hunk 2]
```diff
@@ -22,3 +22,10 @@ GC.gc(true)
 ```
 
 Avoid assuming a key exists in a `WeakKeyIdDict`, especially when allocation occurs. Keys can be removed any time garbage collection occurs. Favor the methods `get!` and `get` over `getindex`.
+
+
+## Credits
+
+Much of the credit for the implementation of WeakKeyIdDict goes to Maura Werder (@mauro3) for the initial [work in the julia repository](https://github.com/JuliaLang/julia/pull/28182), and [DataStructures](https://github.com/JuliaCollections/DataStructures.jl/pull/402). Max Horn (@fingolfin) then updated and debugged a version merged to [AlbstractAlgebra](https://github.com/Nemocas/AbstractAlgebra.jl/pull/1419).
+
+The version here adds a few additional tests for edge-cases, fixing related bugs along the way, that arose from work in base [julia](https://github.com/JuliaLang/julia).
```
Comment (ericphanson) [inline README.md:31]:
```suggestion
The version here adds a few additional tests for edge-cases, fixing related bugs along the way, using code from some of the `IdDict` tests from Base [julia](https://github.com/JuliaLang/julia).
```

File: LICENSE.md (hunk 1 above)
Comment (haberdashPI) [reply LICENSE.md:23]: I changed this based on my understand of @fingolfin's comments here https://github.com/Nemocas/AbstractAlgebra.jl/issues/1666#issuecomment-2052081190, that

> so it would be fine to have just the MIT license in your repository

Comment (ericphanson) [reply LICENSE.md:23]: ah ok, I missed that bit

File: README.md [hunk 3]
```diff
@@ -22,3 +22,10 @@ GC.gc(true)
 ```
 
 Avoid assuming a key exists in a `WeakKeyIdDict`, especially when allocation occurs. Keys can be removed any time garbage collection occurs. Favor the methods `get!` and `get` over `getindex`.
+
+
+## Credits
+
+Much of the credit for the implementation of WeakKeyIdDict goes to Maura Werder (@mauro3) for the initial [work in the julia repository](https://github.com/JuliaLang/julia/pull/28182), and [DataStructures](https://github.com/JuliaCollections/DataStructures.jl/pull/402). Max Horn (@fingolfin) then updated and debugged a version merged to [AlbstractAlgebra](https://github.com/Nemocas/AbstractAlgebra.jl/pull/1419).
```
Comment (ericphanson) [inline README.md:29]:
```suggestion
Much of the credit for the implementation of WeakKeyIdDict goes to Maura Werder (@mauro3) for the initial [work in the julia repository](https://github.com/JuliaLang/julia/pull/28182), and [DataStructures](https://github.com/JuliaCollections/DataStructures.jl/pull/402). Max Horn (@fingolfin) then updated and debugged a version merged to [AbstractAlgebra](https://github.com/Nemocas/AbstractAlgebra.jl/pull/1419).
```

**Review by ericphanson — APPROVED** [review]

**Review by fingolfin — APPROVED** [review]
Thank you, this looks good to me :-)

File: README.md (hunk 3 above)
Comment (lgoettgens) [inline README.md:29]:
```suggestion
Much of the credit for the implementation of WeakKeyIdDict goes to Mauro Werder (@mauro3) for the initial [work in the julia repository](https://github.com/JuliaLang/julia/pull/28182), and [DataStructures](https://github.com/JuliaCollections/DataStructures.jl/pull/402). Max Horn (@fingolfin) then updated and debugged a version merged to [AlbstractAlgebra](https://github.com/Nemocas/AbstractAlgebra.jl/pull/1419).
```
At least that's the way the name is written on their webpage https://maurow.bitbucket.io/
